configure.ac

```diff
@@ -507,6 +507,37 @@ AC_MSG_CHECKING([sould NCP support UART])
 AC_MSG_RESULT([${OPENTHREAD_ENABLE_UART}])
 
 #
+# Single Instance Optimization
+#
+
+AC_ARG_ENABLE(single-instance-opt,
```

Comment:
Good point! I don't think we have a current/immediate use-case for this. The only thing that may be easier with optimization disabled are some of the unit tests (e.g., timer unit test). But I think there should be ways to make it work.

I am fine with combining the options into one: 

- `OPENTHREAD_ENABLE_MULTIPLE_INSTANCES` (controlled with `--enable-multiple-instances`): If enabled, we support multiple instances, otherwise we support single instance with optimization always enabled in single instance case.